pkg/workloads/cortex/lib/type/api.py

```diff
@@ -158,3 +160,18 @@ def prediction_metrics(self, dimensions, prediction_value):
                 "Dimensions": dimensions,
                 "Value": float(prediction_value),
             }
+
+
+def get_spec(provider, storage, cache_dir, spec_path):
+    if provider == "local":
+        return read_msgpack(spec_path)
+
+    local_spec_path = os.path.join(cache_dir, "api_spec.msgpack")
```

Comment:
We should check if `api_spec.msgpack` doesn't exist before downloading, otherwise just read from local_spec_path. This should have been there before your PR. That check is even more useful now because the api spec is already downloaded in start_uvicorn

Reply:
Got it. Makes sense.